src/components/dialogs/upload-case.js

```diff
@@ -6,20 +6,68 @@
  */
 
 import { useDispatch, useSelector } from 'react-redux';
-import { selectFile } from '../../redux/actions';
+import {
+    selectFile,
+    setFormatWithParameters,
+    setTempCaseUuid,
+    setformatInvalidMsgError,
+} from '../../redux/actions';
 import Button from '@mui/material/Button';
 import { FormattedMessage } from 'react-intl';
-import React from 'react';
-
+import React, { useState } from 'react';
+import {
+    deleteCaseByCaseUuid,
+    getCaseImportParameters,
+    getCaseUuidWhenUploadFile,
+} from '../../utils/rest-api';
+import CircularProgress from '@mui/material/CircularProgress';
 export const UploadCase = () => {
     const dispatch = useDispatch();
     const selectedFile = useSelector((state) => state.selectedFile);
-
+    const tempCaseUuid = useSelector((state) => state.tempCaseUuid);
+    const [loadingUploadFile, setLoadingUploadFile] = useState(false);
+    const INVALID_FORMAT = 'Invalid format';
     const handleFileUpload = (e) => {
         e.preventDefault();
         let files = e.target.files;
         if (files.size === 0) dispatch(selectFile(null));
-        else dispatch(selectFile(files[0]));
+        else {
+            if (tempCaseUuid != null) {
+                deleteCaseByCaseUuid(tempCaseUuid);
+            }
+            setLoadingUploadFile(true);
+            getCaseUuidWhenUploadFile(files[0])
+                .then((caseUuid) => {
+                    if (caseUuid) {
+                        dispatch(setformatInvalidMsgError(null));
+                        dispatch(selectFile(files[0]));
+                        setLoadingUploadFile(false);
+                        dispatch(setTempCaseUuid(caseUuid));
+                        getCaseImportParameters(caseUuid)
+                            .then((result) => {
+                                result.parameters = result.parameters?.map(
+                                    (p) => {
+                                        let sortedPossibleValue =
+                                            p.possibleValues?.sort((a, b) =>
+                                                a.localeCompare(b)
+                                            );
+                                        p.possibleValues = sortedPossibleValue;
+                                        return p;
+                                    }
+                                );
+                                dispatch(
+                                    setFormatWithParameters(result.parameters)
+                                );
+                            })
+                            .catch(() => dispatch(setFormatWithParameters([])));
```

Comment:
shoudn't we print an error message in addition of setting parameters to empty ?

Reply:
on the snackbar i mean

Reply:
i think in the old behaviour there wasn't en error to print on the snackbar